os_migrate/plugins/module_utils/resource.py

```diff
@@ -0,0 +1,236 @@
+        current_trimmed = _trim_info(current)
+        target_trimmed = _trim_info(target)
+        return current_trimmed != target_trimmed
+
+    def _data_from_sdk_and_refs(self, sdk_res, refs):
+        # we could move these methods from serialization.py to private
+        # under Resource
+        set_ser_params_same_name(params, sdk_res, self.params_from_sdk)
+        set_ser_params_same_name(params, refs, self.params_from_sdk)
+        set_ser_params_same_name(info, sdk_res, self.info_from_sdk)
+        set_ser_params_same_name(info, refs, self.info_from_refs)
+
+    def _sdk_params_from_params_and_refs(self, sdk_params, refs):
+        # we could move these methods from serialization.py to private
+        # under Resource
+        set_sdk_params_same_name(self.params(), sdk_params, self.sdk_params_from_params)
+        set_sdk_params_same_name(refs, sdk_params, self.sdk_params_from_refs)
+
+    def type(self):
+        return self.data[const.RES_TYPE]
+
+    def params(self):
+        return self.data[const.RES_PARAMS]
+
+    def info(self):
+        return self.data[const.RES_INFO]
+
+    def params_and_info(self):
+        return (self.params(), self.info())
+
+    def _sort_param(self, param_name):
+        self.params()[param_name] = sorted(self.params()[param_name])
+
+    def _sort_param(self, param_name):
+        self.info()[param_name] = sorted(self.info()[param_name])
+
+
+# This would obviously go to network.py but i want to keep the "idea
+# pull request" simple.
+class Network(Resource):
+
+    params_from_sdk = [
+        'description',
+        'dns_domain',
+        'is_admin_state_up',
+        'is_default',
+        'is_port_security_enabled',
+        'is_router_external',
+        'is_shared',
+        'is_vlan_transparent',
+        'mtu',
+        'name',
+        'provider_network_type',
+        'provider_physical_network',
+        'provider_segmentation_id',
+        'segments',
+    ]
+    params_from_refs = [
+        'qos_policy_name',
+    ]
+    info_from_sdk = [
+        'availability_zones',
+        'created_at',
+        'id',
+        'project_id',
+        'qos_policy_id',
+        'revision_number',
+        'status',
+        'updated_at',
+    ]
+
+    sdk_params_from_params = [
+        'description',
+        'dns_domain',
+        'is_admin_state_up',
+        'is_default',
+        'is_port_security_enabled',
+        'is_router_external',
+        'is_shared',
+        'is_vlan_transparent',
+        'mtu',
+        'name',
+        'provider_network_type',
+        'provider_physical_network',
+        'provider_segmentation_id',
+        'segments',
+    ]
+    sdk_params_from_refs = [
+        'qos_policy_id',
+    ]
+
+
+    @classmethod
+    def from_sdk(cls, sdk_resource, conn):
+        # >>> Internal composition over inheritance! <<<
+        # This is more future-proof, composable, less restrictive
+        # approach than calling the same method via super.
+        obj = cls()
```

Comment:
This becomes the new bolierplate.  It's going to be the implementation for most if not all.  If it's the common case, why not have this be the default method and override IF you are planning to deviate from it?